src/components/sets/CellSetsManagerSubscriber.js

```diff
@@ -93,16 +93,17 @@ export default function CellSetsManagerSubscriber(props) {
 
   // Get data from loaders using the data hooks.
   const [cells] = useCellsData(loaders, dataset, setItemIsReady, addUrl, true);
-  const [cellSets] = useCellSetsData(loaders, dataset, setItemIsReady, addUrl, true, (data) => {
-    if (data && data.tree.length >= 1) {
-      // eslint-disable-next-line no-underscore-dangle
-      const newAutoSetSelectionKeys = data.tree[0].children.map(node => node._state.key);
-      const newAutoSetSelections = treeToSetNamesByKeys(data, newAutoSetSelectionKeys);
-      setAutoSetSelections(newAutoSetSelections);
-    } else {
-      setAutoSetSelections([]);
-    }
-  });
+  const [cellSets] = useCellSetsData(loaders, dataset, setItemIsReady, addUrl, true,
+    (data) => {
+      if (data && data.tree.length >= 1) {
+        // eslint-disable-next-line no-underscore-dangle
+        const newAutoSetSelectionKeys = data.tree[0].children.map(node => node._state.key);
+        const newAutoSetSelections = treeToSetNamesByKeys(data, newAutoSetSelectionKeys);
+        setAutoSetSelections(prev => ({ ...prev, [dataset]: newAutoSetSelections }));
```

Comment:
Without testing in the portal, I think this might fail on switching datasets since [this line](https://github.com/hubmapconsortium/vitessce/pull/756/files#diff-15aff9a793c3ad2d7bfa9236a30d8c31R90) sets the `prev` to `undefined`.  That's why I wrote my update functions the way I did, with the spread on the inside.

Reply:
Oh yes thanks for catching that.